### ISSUE TITLE
Can't use dependencies inherited from remote layer inside app

### ISSUE THREAD

**Issue by mrleblanc101** [nuxt/nuxt#30103]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v20.18.0
- Nuxt Version:     3.14.1592
- CLI Version:      3.16.0
- Nitro Version:    2.10.4
- Package Manager:  pnpm@9.14.2
- Builder:          -
- User Config:      default
- Runtime Modules:  @nuxt/eslint
- Build Modules:    -
------------------------------

### Reproduction

Layer: https://stackblitz.com/~/github.com/mrleblanc101/repro-dependencies-layer
App: https://stackblitz.com/~/github.com/mrleblanc101/repro-dependencies-app

I don't think we can test in StackBlitz because of `c12 / giget` not working there, so here are the Github repo
Layer: https://github.com/mrleblanc101/repro-dependencies-layer
App: https://github.com/mrleblanc101/repro-dependencies-app

### Describe the bug

In the app, when running `pnpm install && pnpm run dev`, the app dependencies are installed and the layers dependencies are also installed.
See tailwind-merge is installed properly from the layer:
![Image](https://github.com/user-attachments/assets/ad9b2cf9-e60f-4644-b3f2-b4504c98caf8)

But in VS Code, the dependencies installed by the layer are not recognized by `ts-plugin`, but the `twMerge` package still work: 
![Image](https://github.com/user-attachments/assets/b09547ff-8163-40ec-9006-fd26b0e9adc4)

I have a similar problem with other dependencies.
See eslint-config-prettier is installed properly from the layer:
![Image](https://github.com/user-attachments/assets/265fa00f-b940-4dbb-91ea-b05ca4023be5)

Also, there is an error in the `pnpm run dev`, but it doesn't crash and the app still render:
![Image](https://github.com/user-attachments/assets/c76c042e-bdcd-465e-81e9-941cbeeeb30c)

I think this is because `tailwind-merge` and `eslint-config-prettier` are installed inside `node_modules/.c12/layer_name/node_modules` and not at the root of the `node_modules`.
I have tried with and without `shamefully-hoist=true` in `.npmrc` inside the app and the layer.
I also made sure that the packages are installed inside `dependencies` and not `devDependencies` because the later would then not be installed inside the app in the postinstall `nuxi prepare`
![Image](https://github.com/user-attachments/assets/34c43763-dfbd-4df9-84bf-83a537d59980)

### Logs

```shell-script

```

**Comment by mrleblanc101** [comment]
As I didn't want to publish my layer as a package on our private Gitlab instance, I found a workaround instead of using Giget:
`pnpm install git+ssh://git@gitlab.<instance>.com/<group_name>/<project_name>` which properly install the sub-dependencies.

**Comment by mrleblanc101** [comment]
Just realized that this won't work inside our CD/CI, I need to switch to git+https and use deploy token

**Comment by danielroe** [comment]
I think this is likely not a Nuxt issue - the problem here is that _eslint_ can't resolve those dependencies - linters and similar non-Nuxt dependencies have no ability to work with Nuxt remote layers...

Dependencies of the layer itself work fine within your Nuxt code, right?

**Comment by danielroe** [comment]
That is expected.

The type issue we could probably resolve (maybe hoist types for all layer dependencies?)

But the layer simply isn't a dependency of your project if it's using giget - Nuxt does some magic to make it work with _Nuxt's_ code but we can't solve prettier or eslint's module resolution.

**Comment by mrleblanc101** [comment]
@danielroe Maybe this should be better explained inside the [Authoring Nuxt Layers] guide (https://nuxt.com/docs/guide/going-further/layers) ?
I'm not sure why giget would be prefered instead of something like `pnpm install git+ssh://git@gitlab.<instance>.com/<group_name>/<project_name>`

**Comment by danielroe** [comment]
`giget` remote sources aren't _preferred_ - they just enables usage that might not be otherwise possible.

In your case I would make the layer an explicit dependency so that your package manager can handle installing eslint plugins etc.

**Comment by mrleblanc101** [comment]
@danielroe I see now, thanks. 
I guess I jumped 1 step ahead, I knew I didn't want to publish my layer to npm since it's private.
The alternative is to publish the layer to the private Gitlab npm package registery, but that requires some setup.
So when I saw this part in the documentation, and the callout about private Gitlab instance, I jumped to the conclusion that this was the solution:
![Image](https://github.com/user-attachments/assets/dc81cda0-790a-419f-b0de-ea468963d3af)
But with npm, we can require a Git repo `.git` as a dependency without having to publish the repo to the private Gitlab npm package registery.

We should find a way to highlight the advantages and limitation of giget.
I just saw this in the doc, add tought, neat we can link a remote layer directly.
Then I wondered how that would work with a private gitlab instance, and the callout
